misk/src/main/kotlin/misk/web/actions/AdminTabAction.kt

```diff
@@ -0,0 +1,41 @@
+package misk.web.actions
+
+import misk.security.authz.Unauthenticated
+import misk.web.Get
+import misk.web.RequestContentType
+import misk.web.ResponseContentType
+import misk.web.mediatype.MediaTypes
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class AdminTabAction : WebAction {
+  @Inject lateinit var registeredTabs: List<AdminTab>
+
+  @Get("/api/_admin/loader/all")
+  @RequestContentType(MediaTypes.APPLICATION_JSON)
+  @ResponseContentType(MediaTypes.APPLICATION_JSON)
+  @Unauthenticated
+  fun getAll(): Response {
+    return Response(tabs = registeredTabs.map { tab -> tab.slug to tab }.toMap())
+  }
+
+  data class Response(val tabs: Map<String, AdminTab>)
+}
+
+data class AdminTab(
+  val name: String,
+  val slug: String,
+  val url_path_prefix: String,
+  val icon: String = "widget-button"
+) {
+  init {
+    require(name.first().isUpperCase() &&
```

Comment:
for things like isUpperCase I prefer plain documentation vs. runtime enforcement. This kind of thing tends to become too strict in the weeds. (for example, our “iOS Apps” tab).

Reply:
Fair point and good edge case with the `iOS Apps`, I'll remove the syntax enforcement requirements and add documentation.